client/src/pages/ProfileListings/ProfileListings.tsx

```diff
@@ -0,0 +1,164 @@
+    if (dateFrom instanceof Date && dateTo instanceof Date) {
+      setDateRange({
+        from: dateFrom,
+        to: dateTo,
+      });
+      const formattedDateFrom = moment(dateFrom).format('MMMM Do YYYY');
+      const formattedDateTo = moment(dateTo).format('MMMM Do YYYY');
+      setFormattedDateRange(`${formattedDateFrom} – ${formattedDateTo}`);
+    } else {
+      setFormattedDateRange(`any`);
+    }
+  };
+
+  const updateSearch = () => {
+    const searchResults: User[] = [];
+    setDisplayedUsers([]);
+    users.map((user: User) => {
+      if (user.city.toLowerCase().includes(search)) {
+        searchResults.push(user);
+        setDisplayedUsers(searchResults);
+      }
+    });
+  };
+
+  //   Updates based on date selection
+  useEffect(() => {
+    if (formattedDateRange === 'any') {
+      setDisplayedUsers(users.slice(0, 6));
+    } else {
+      if (dateRange.from && dateRange.to) {
+        const searchResults: User[] = [];
+        setDisplayedUsers([]);
+        const range = moment.range(dateRange.from, dateRange.to);
+        users.map((user) => {
+          //   If at least one available date falls within the selected range, the user
+          //   is added to the displayedUsers array.
+          const isInDateRange = (date: Date) => {
+            return range.contains(date);
+          };
+          if (user.availableDates.some(isInDateRange)) {
+            searchResults.push(user);
+            setDisplayedUsers(searchResults);
+          }
+        });
+      }
+    }
+  }, [formattedDateRange, setFormattedDateRange]);
+
+  //   Updates based on search parameters
+  useEffect(() => {
+    if (search) {
+      updateSearch();
+    } else {
+      setDisplayedUsers(users.slice(0, 6));
+    }
+  }, [search, setSearch]);
+
+  const profileCardGrid = (
+    <Grid container className={classes.profilesContainer} xs={9}>
+      {displayedUsers.map((user) => (
+        <ProfileCard user={user} key={user._id} />
+      ))}
+      <Snackbar open={snackbarOpen} autoHideDuration={2000} onClose={handleSnackbarClose}>
+        <Alert severity="info">No more users found</Alert>
+      </Snackbar>
+    </Grid>
+  );
+
+  return (
+    <Grid container className={classes.root}>
+      <Typography variant="h4" className={classes.title}>
+        Your search results
+      </Typography>
+      <Grid>
+        <Grid container className={classes.searchDateContainer}>
+          <Grid className={classes.search}>
+            <SearchIcon className={classes.searchIcon} />
+            <InputBase
+              placeholder="Search by city…"
+              inputProps={{ 'aria-label': 'search' }}
+              onChange={(e) => {
+                setSearch(e.target.value);
+              }}
+            />
+          </Grid>
+          <Grid>
+            <Divider orientation="vertical" />
+          </Grid>
+          <Grid className={classes.dateRangeContainer}>
+            <Button onClick={handleCalendarOpen}>
```

Comment:
```suggestion
            <Button onClick={handleCalendarOpen} component="span">
```
By default, material-ui use `button` tag. `button` tag should have some content as per WCAG guidelines [https://www.w3.org/WAI/standards-guidelines/act/rules/button-non-empty-accessible-name-97a4e1/](https://www.w3.org/WAI/standards-guidelines/act/rules/button-non-empty-accessible-name-97a4e1/)

Reply:
I see what you're saying, but I'm afraid I don't understand why component="span" should be added. Wouldn't it be more effective to add aria-label="button", in order to assign an accessible name to the element? Is there a reason why component="span" would be preferable?

Reply:
The thing if you even provide `aria-label`, it will not solve problem. As per WCAG guidelines, button tag expects some content like a words or string in it. You are providing icon into button tag which is not a word or string. So that's why I suggest to change this to span tag.

Reply:
Oh yes, I see what you mean.